pkg/cmd/adm/register_member_test.go

```diff
@@ -589,104 +412,109 @@ func TestRegisterMember(t *testing.T) {
+							Status: corev1.ConditionTrue,
+						},
+					},
+				}
+			}
+		}
+		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
-func TestRunAddClusterScriptFailed(t *testing.T) {
-	// given
-	SetFileConfig(t, Host(), Member())
-	hostKubeconfig := PersistKubeConfigFile(t, HostKubeConfig())
-	memberKubeconfig := PersistKubeConfigFile(t, MemberKubeConfig())
-	gock.New(AddClusterScriptDomain).
-		Get(AddClusterScriptPath).
-		Persist().
-		Reply(404)
-	defer gock.OffAll()
-
-	for _, clusterType := range configuration.ClusterTypes {
-		t.Run("for cluster name: "+clusterType.String(), func(t *testing.T) {
-			// given
-			expArgs := []string{"--type", clusterType.String(), "--host-kubeconfig", hostKubeconfig, "--host-ns", "host-ns", "--member-kubeconfig", memberKubeconfig, "--member-ns", "member-ns", "--lets-encrypt"}
-			ocCommandCreator := NewCommandCreator(t, "echo", "bash",
-				AssertFirstArgPrefixRestEqual("(.*)/add-cluster-(.*)", expArgs...))
-			term := NewFakeTerminalWithResponse("Y")
-
-			// when
-			err := runAddClusterScript(term, ocCommandCreator, clusterType, hostKubeconfig, "host-ns", memberKubeconfig, "member-ns", "", true)
-
-			// then
-			require.Error(t, err)
-			assert.NotContains(t, term.Output(), "Command to be called")
-		})
+func mockCreateToolchainClusterWithReadyCondition(fakeClient *test.FakeClient) {
+	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
+		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
+			obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
+				Conditions: []toolchainv1alpha1.Condition{
+					{
+						Type:   toolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			}
+		}
+		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
+func verifyToolchainClusterSecret(t *testing.T, fakeClient *test.FakeClient, secretNamespace, accessNamespace, tcName string) {
+	secrets := &corev1.SecretList{}
+	require.NoError(t, fakeClient.List(context.TODO(), secrets, runtimeclient.InNamespace(secretNamespace)))
+	assert.Len(t, secrets.Items, 1)
+	assert.NotEmpty(t, secrets.Items[0].Labels)
+	assert.Equal(t, tcName, secrets.Items[0].Labels[toolchainv1alpha1.ToolchainClusterLabel])
+	assert.NotEmpty(t, secrets.Items[0].StringData["token"])
+	assert.NotEmpty(t, secrets.Items[0].StringData["kubeconfig"])
+	apiConfig, err := clientcmd.Load([]byte(secrets.Items[0].StringData["kubeconfig"]))
+	require.NoError(t, err)
+	require.False(t, api.IsConfigEmpty(apiConfig))
+	assert.Equal(t, "https://cool-server.com", apiConfig.Clusters["cluster"].Server)
+	assert.True(t, apiConfig.Clusters["cluster"].InsecureSkipTLSVerify) // by default the insecure flag is being set
+	assert.Equal(t, "cluster", apiConfig.Contexts["ctx"].Cluster)
+	assert.Equal(t, accessNamespace, apiConfig.Contexts["ctx"].Namespace)
+	assert.NotEmpty(t, apiConfig.AuthInfos["auth"].Token)
```

Comment:
the same

Reply:
done in https://github.com/kubesaw/ksctl/pull/53/commits/6503e6d43fa3d5f71ef22e2da0e681346b836280